app/public/views/periods/overview/dataExport.js

```diff
@@ -0,0 +1,156 @@
+    all: 'all',
+    currentPage: 'currentPage',
+    selected: 'selected',
+    notSelected: 'notSelected',
+    visible: 'visible',
+};
+
+export default function dataExport(close, dataModel) {
+    const pageName = dataModel.name;
+    const rowsPreferenceSelectId = 'rows-preference-selection';
+    const columnsPreferenceSelectId = 'columns-preference-selection';
+    const exportFormatSelectId = 'export-format-selection';
+
+    const title = h('h3.text-primary', `${pageName} export`);
+
+    const exportData = async () => {
+        const remoteData = dataModel.currentPageData;
+
+        remoteData.match({
+            NotAsked: () => {},
+            Loading: () => {},
+            Success: async () => {
+                await dataModel.createDataExport(remoteData.payload, selectedDataFormat());
+            },
+            Failure: () => {},
+        });
+        close();
+    };
+
+    const handleRowsPreferenceSelection = () => {
+        const exportPreferenceSelection = document.getElementById(rowsPreferenceSelectId);
+        const selectedPreference = exportPreferenceSelection.options[exportPreferenceSelection.selectedIndex].value;
+        switch (selectedPreference) {
+            case exportPreferences.all:
+                /* */
+                break;
+            case exportPreferences.currentPage:
+                /* */
+                break;
+            case exportPreferences.selected:
+                /* */
+                break;
+            case exportPreferences.notSelected:
+                /* */
+                break;
+            case exportPreferences.visible:
+                /* */
+                break;
+            default:
+                break;
+        }
+    };
+
+    const handleColumnsPreferenceSelection = () => {
+        const exportPreferenceSelection = document.getElementById(columnsPreferenceSelectId);
+        const selectedPreference = exportPreferenceSelection.options[exportPreferenceSelection.selectedIndex].value;
+        switch (selectedPreference) {
+            case exportPreferences.all:
+                /* */
+                break;
+            case exportPreferences.selected:
+                /* */
+                break;
+            case exportPreferences.notSelected:
+                /* */
+                break;
+            case exportPreferences.visible:
+                /* */
+                break;
+            default:
+                break;
+        }
+    };
+
+    const selectedDataFormat = () => {
+        const exportFormatSelection = document.getElementById(exportFormatSelectId);
+        const selectedFormat = exportFormatSelection.options[exportFormatSelection.selectedIndex].value;
+        switch (selectedFormat) {
+            case exportFormats.csv:
+                return selectedFormat;
+            case exportFormats.json:
+                return selectedFormat;
+            default:
+                return selectedFormat.csv;
+        }
```

Comment:
If it is not strictly necessary for those lambdas to be inside of main function body, they should be moved outside

Reply:
Fine, fixed that as well.